sdk/core/System.ClientModel/src/Pipeline/ClientPipeline.cs

```diff
@@ -129,8 +133,11 @@ public static ClientPipeline Create(ClientPipelineOptions? options = default)
 
         int perCallIndex = index;
 
-        // Add retry policy.
-        policies[index++] = options.RetryPolicy ?? ClientRetryPolicy.Default;
+        if (options.RetryPolicy is not null)
+        {
+            // Add retry policy.
+            policies[index++] = options.RetryPolicy;
```

Comment:
One disadvantage of this approach is that, if the user isn't going to use the default policies, we're still allocating them to set them on ClientPipelineOptions.